File: nj2k/testData/newJ2k/issues/kt-21189.java
public class ArrayInitializerBugKt {

    private static final byte[] GREETING = { 'H', 'e', 'l', 'l', 'o', ',', ' ', 'b', 'u', 'g', '!' };

    public static void main(String... args) {
        String greeting = new String(GREETING);
        System.out.println(greeting);
    }

}

File: nj2k/testData/newJ2k/issues/kt-21189.kt
object ArrayInitializerBugKt {
    private val GREETING = byteArrayOf(
        'H'.toByte(),
        'e'.toByte(),
        'l'.toByte(),
        'l'.toByte(),
        'o'.toByte(),
        ','.toByte(),
        ' '.toByte(),
        'b'.toByte(),
        'u'.toByte(),
        'g'.toByte(),
        '!'.toByte()
    )

    fun main(vararg args: String?) {
        val greeting = String(GREETING)
        println(greeting)
    }
}